tests/php/ORM/DataObjectTest.php

```diff
@@ -2536,4 +2536,26 @@ public function testShallowRecursiveWrite()
             'Root is 2 step remove from grand children. It was not written on a shallow recursive write.'
         );
     }
+
+    /**
+     * @param string $identifier
+     * @param string $class
+     * @param string $expected
+     * @dataProvider uniqueKeysProvider
+     */
+    public function testUniqueKey(string $identifier, string $class, string $expected): void
+    {
+        $object = $this->objFromFixture($class, $identifier);
+        $this->assertEquals($expected, $object->getUniqueKey());
+    }
+
+    public function uniqueKeysProvider(): array
```

Comment:
If the entire point of this PR is to allow Fluent or Versioned to alter the caching key of some objects, then I think we should have a test case with an dummy extension that does that.

Reply:
good point, will add it to the test